pkg/cmdutil/config.go

```diff
@@ -34,17 +33,17 @@ func GetFullConfig(defaultConfig *types.Config, clusterConfig *types.ClusterConf
 	return &ndf, &cc, nil
 }
 
-// ensureBootImageVersion patches the boot image tag as necessary based on the desired version
-// of Kubernetes. It returns the updated image string.
-func EnsureBootImageVersion(requestedVersion string, image string) string {
+// EnsureBootImageVersion appends an image tag consisting of the Kubernetes version ,if the image string does not currently have a tag.
+// It returns the updated image string.
+func EnsureBootImageVersion(kubeVersion string, image string) string {
 	// if the user already specified a tag at the end of the image, use that tag and return
 	parts := strings.Split(image, ":")
-	_, err := semver.NewVersion(parts[len(parts)-1])
-	if err == nil {
+	// This means that the image has at least two colons, one for the transport and another for the tag
+	if len(parts) > 2 {
```

Comment:
It is also possible to have a colon for the port, as in "docker://myregistry:8443/ocne/ock:blah".  That needs to be accounted for.

Reply:
Ok! I will make this change and test it.

Reply:
@George-Aeillo it took some time but I wrote and tested a new function to extract image tags reliably. I addressed the transport related issues we discussed offline. Can I push the function to your branch?

Reply:
Please check this logic and try accordingly https://github.com/oracle-cne/ocne/blob/main/pkg/image/image.go#L404-L413
```
named = reference.TagNameOnly(named)
tagged, ok := named.(reference.NamedTagged)
if !ok {
	tmp := fmt.Sprintf("error removing image tag from string %s", image)
	return image, errors.New(tmp)
}
justImage, found := strings.CutSuffix(named.String(), ":"+tagged.Tag())
if found {
	return justImage, nil
}
```

Reply:
Ok! Thanks @prasad-shirodkar

Reply:
Pushed it, please see https://github.com/oracle-cne/ocne/blob/0fcc0b15e4365eb6319264c95b9b6871174654ca/pkg/image/image.go#L396

Reply:
I used Zaid's function to make my check more robust and tested it with the same two use cases that I originally tested with before (one test without the image tag, and one test with a tag)